core/src/main/scala/org/apache/spark/util/Benchmark.scala

```diff
@@ -124,7 +124,7 @@ private[spark] object Benchmark {
     }
     val best = runTimes.min
     val avg = runTimes.sum / iters
-    Result(avg / 1000000, num / (best / 1000), best / 1000000)
+    Result(avg / 1000000, num.toDouble / (best / 1000), best / 1000000)
```

Comment:
We need to keep the precision here.  The `bestMs/avgMs` can be well controlled in an appropriate number, but the `rate` can't. And we use `rate` as a divisor later, so if `rate` is small(assume we are benchmarking some slow operations), we will get large deviation. BTW, we use `%10.1f` to print `rate` but previously `rate` is always integral.

cc @davies